pyro/infer/importance.py

```diff
@@ -44,3 +44,33 @@ def _traces(self, *args, **kwargs):
                 poutine.replay(self.model, trace=guide_trace)).get_trace(*args, **kwargs)
             log_weight = model_trace.log_prob_sum() - guide_trace.log_prob_sum()
             yield (model_trace, log_weight)
+
+    def get_log_normalizer(self):
+        """
+        Estimator of the normalizing constant of the target distribution.
+        (mean of the unnormalized weights)
+        """
+        log_w = torch.tensor(self.log_weights)
+        log_num_samples = torch.log(torch.tensor(self.num_samples, dtype=torch.float))
```

Comment:
this `dtype` be the type of `log_w`

Reply:
Can you please elaborate? 

self.num_samples is an int by default; log_w is a tensor of floats.

Reply:
i was thinking if default_tensor_type is `torch.DoubleTensor` then `log_w` will be a `torch.DoubleTensor`, but `log_num_samples` is a `torch.FloatTensor` so the following line will throw an error like `Expected object of type torch.DoubleTensor but found type torch.FloatTensor`.

Reply:
It isn't throwing an error, but its an easy fix. Will change it to be the same type.

Reply:
have you tried changing the default tensor type? on my system (pytorch 0.4.0) i get an error:

```
In [3]: torch.set_default_tensor_type('torch.DoubleTensor')

In [4]: log_w = torch.tensor([0.1])

In [5]: log_num_samples = torch.log(torch.tensor(2, dtype=torch.float))

In [6]: log_w - log_num_samples
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-6-1e3c76f98fb1> in <module>()
----> 1 log_w - log_num_samples

RuntimeError: Expected object of type torch.DoubleTensor but found type torch.FloatTensor for argument #3 'other'
```

Reply:
I'll look into it.

Reply:
I tested with PyTorch 1.0.0rc1 and didn't get any issues, 0.4.0 must be less forgiving with types. 

Anyways, this was easily fixed to be the default tensor type by simply multiplying num_samples by 1. in the latest commit.